code/controllers/subsystems/throwing.dm

```diff
@@ -150,18 +148,11 @@ SUBSYSTEM_DEF(throwing)
 		AM.Move(step, get_dir(AM, step))
 
 		if (!AM.throwing) // we hit something during our move
-			finalize(hit = TRUE)
```

Comment:
No finalize if it stops being thrown?

Reply:
it should finalise upon bumping into anything. This is just to prevent the callback proc and finalize being called multiple times.